zone/mob.cpp

```diff
@@ -2838,8 +2838,44 @@ void Mob::SendStunAppearance()
 	safe_delete(outapp);
 }
 
-void Mob::SendAppearanceEffect(uint32 parm1, uint32 parm2, uint32 parm3, uint32 parm4, uint32 parm5, Client *specific_target){
+void Mob::SendAppearanceEffect(uint32 parm1, uint32 parm2, uint32 parm3, uint32 parm4, uint32 parm5, Client *specific_target, 
+	uint32 value1slot, uint32 value1ground, uint32 value2slot, uint32 value2ground, uint32 value3slot, uint32 value3ground, 
+	uint32 value4slot, uint32 value4ground, uint32 value5slot, uint32 value5ground){
 	auto outapp = new EQApplicationPacket(OP_LevelAppearance, sizeof(LevelAppearance_Struct));
+	
+	/* Location of the effect from value#slot, this is removed upon mob death/despawn.
+		0 = pelvis1
+		1 = pelvis2
+		2 = helm
+		3 = Offhand
+		4 = Mainhand
+		5 = left foot
+		6 = right foot
+		9 = Face
+
+		value#ground = 1, will place the effect on ground, of corresponding slot is set to 0 effect is permanenant, if > 0 will fade if mob death/despawn.
+	*/
+
+	//higher values can crash client
+	if (value1slot > 9) {
+		value1slot = 1;
+	}
+	else if (value2slot > 9) {
+		value1slot = 1;
+	}
+	else if (value2slot > 9) {
+		value1slot = 1;
+	}
+	else if (value3slot > 9) {
+		value1slot = 1;
+	}
+	else if (value4slot > 9) {
+		value1slot = 1;
+	}
+	else if (value5slot > 9) {
+		value1slot = 1;
+	}
+
```

Comment:
Should these be just ifs? So that they’re all checked versus only fixing the first one it finds?

Reply:
Also looks to just be setting value1 to 1 if value2/3/4/5 fails.